loopout.go

```diff
@@ -1101,121 +1102,51 @@ func (s *loopOutSwap) waitForHtlcSpendConfirmed(globalCtx context.Context,
-							"state")
-					}
-
-					return nil, nil
-				}
-
-				// When using taproot HTLCs we're pushing the
-				// preimage before attempting to sweep. This
-				// way the server will know that the swap will
-				// go through and we'll be able to MuSig2
-				// cosign our sweep transaction. In the worst
-				// case if the server is uncooperative for any
-				// reason we can still sweep using scriptpath
-				// spend.
-				err = s.setStatePreimageRevealed(ctx)
+			// sweepConfTarget will return false if the preimage is
+			// not revealed yet but the conf target is closer than
+			// 20 blocks. In this case to be sure we won't attempt
+			// to sweep at all and we won't reveal the preimage
+			// either.
+			_, canSweep := s.sweepConfTarget()
+			if !canSweep {
+				s.log.Infof("Aborting swap, timed " +
+					"out on-chain")
+
+				s.state = loopdb.StateFailTimeout
+				err := s.persistState(ctx)
 				if err != nil {
-					return nil, err
-				}
-
-				if !paymentComplete {
-					// Push the preimage for as long as the
-					// server is able to settle the swap
-					// invoice. So that we can continue
-					// with the MuSig2 sweep afterwards.
-					s.pushPreimage(ctx)
+					log.Warnf("unable to persist " +
+						"state")
 				}
 
-				// Now attempt to publish a MuSig2 sweep txn.
-				// Only attempt at most maxMusigSweepRetires
-				// times to still leave time for an emergency
-				// script path sweep.
-				if musigSweepTryCount < maxMusigSweepRetries {
-					success := s.sweepMuSig2(
-						ctx, htlcOutpoint, htlcValue,
-					)
-					if !success {
-						musigSweepTryCount++
-					} else {
-						// Mark that we had a sweep
-						// that was successful. There's
-						// no need for the script spend
-						// now we can just keep pushing
-						// new sweeps to bump the fee.
-						musigSweepSuccess = true
-					}
-				} else if !musigSweepSuccess {
-					// Attempt to script path sweep. If the
-					// sweep fails, we can't do any better
-					// than go on and try again later as
-					// the preimage is already revealed and
-					// the server settled the swap payment.
-					// From the server's point of view the
-					// swap is succeeded at this point so
-					// we are free to retry as long as we
-					// want.
-					err := s.sweep(
-						ctx, htlcOutpoint, htlcValue,
-					)
-					if err != nil {
-						log.Warnf("Failed to publish "+
-							"non-cooperative "+
-							"sweep: %v", err)
-					}
-				}
+				return nil, nil
+			}
 
-				// If the result of our spend func was that the
-				// swap has reached a final state, then we
-				// return nil spend details, because there is
-				// no further action required for this swap.
-				if s.state.Type() != loopdb.StateTypePending {
-					return nil, nil
-				}
-			} else {
-				err := s.sweep(ctx, htlcOutpoint, htlcValue)
-				if err != nil {
-					return nil, err
-				}
+			// Send the sweep to the sweeper.
+			err := s.batcher.AddSweep(globalCtx, sweepReq)
```

Comment:
IIUC the sweep batch will potentially wait for the next block to publish right? Also what happens if we re-add?

Reply:
Yes, we wait for next block in order to create a small buffering effect and help sweeps get batched before publishing our fisrt tx.

If we re-add we simply update the sweep.
e.g We restart, a batch spins up and loads all sweeps from storage. In parallel the swap is resuming and will eventually request a sweep from the batcher, handing over the sweep related details but also the notifier. When the batcher adds that sweep again to the same batch, its fields will be updated, successfully setting the `notifier`

Reply:
> Yes, we wait for next block in order to create a small buffering effect and help sweeps get batched before publishing our first tx.

I think this'll just potentially the user experience worse for the majority of users who only swap once at a time.

Reply:
The fact that the sweep already has a default conf height of `12` blocks kind of lowers the significance of this initial buffering. I'm open to changing, would like to cc @sputn1ck who was behind the idea of this initial buffering

Reply:
Ok so after offline chat: decided to change to a time delay of a few seconds